.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,328 @@
+        run: cmake -BBuild/Release/x64/ -Hsrc/OSPSuite.FuncParserNative/ -DCMAKE_BUILD_TYPE=Release -DCMAKE_OSX_ARCHITECTURES=arm64
+
+      - name: Configure the native build (debug)
+        run: cmake -BBuild/Debug/x64/ -Hsrc/OSPSuite.FuncParserNative/ -DCMAKE_BUILD_TYPE=Debug -DCMAKE_OSX_ARCHITECTURES=arm64
+      
+      - name: Build native libraries (release)
+        run: make -C Build/Release/x64
+
+      - name: Build native libraries (debug)
+        run: make -C Build/Debug/x64
+        
+      - name: Build the project (release)
+        run: dotnet build OSPSuite.FuncParser.sln --configuration Release
+
+      - name: Build the project (debug)
+        run: dotnet build OSPSuite.FuncParser.sln --configuration Debug
+
+      - name: Test
+        run: dotnet test --configuration Release --no-build
+
+      - name: Pack the project
+        run: dotnet pack src/OSPSuite.FuncParser/ -o ./ -p:PackageVersion=${{env.APP_VERSION}}
+
+      - name: Push nupkg as artefact
+        # if it is a push to a branch
+        if: github.event_name == 'push' && github.ref_name != github.event.repository.default_branch
+        uses: actions/upload-artifact@v3
+        with:
+          name: OSPSuite.FuncParser.MacOS.Arm64
+          path: ./*.nupkg
+
+      - name: Publish to GitHub registry
+        # if it is a merge to default branch
+        if: github.event_name == 'push' && github.ref_name == github.event.repository.default_branch
+        run: dotnet nuget push ./*.nupkg --source https://nuget.pkg.github.com/${{github.repository_owner}}/index.json --api-key ${{ secrets.GITHUB_TOKEN }}
+
+  build-macos-x64:
+    runs-on: macos-13
+    needs: get-version
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          submodules: 'true'
+
+      - name: define env variables
+        run: |
+          echo "APP_VERSION=${{needs.get-version.outputs.APP_VERSION}}" >> $GITHUB_ENV
+          echo "BUILD_ID=${{needs.get-version.outputs.BUILD_ID}}" >> $GITHUB_ENV
+
+      - name: Restore dependencies
+        run: dotnet restore --source https://ci.appveyor.com/nuget/ospsuite-utility --source https://ci.appveyor.com/nuget/ospsuite-bddhelper --source https://www.nuget.org/api/v2/
+
+      - name: Update version.h
+        run: |
+          LC_ALL=C sed -i '' -e "s@RELEASE 0@RELEASE ${{env.RELEASE}}@g" \
+                     -e "s@MAJOR 0@MAJOR ${{env.MAJOR}}@g" \
+                     -e "s@MINOR 0@MINOR ${{env.MINOR}}@g" \
+                     -e "s@BUILD 0@BUILD ${{ env.BUILD_ID }}@g" \
+                     src/OSPSuite.FuncParserNative/version.h
+
+      - name: Remove Native Project for macOS
+        run: dotnet sln OSPSuite.FuncParser.sln remove src/OSPSuite.FuncParserNative/OSPSuite.FuncParserNative.vcxproj
+
+      - name: Configure the native build (release)
+        run: cmake -BBuild/Release/x64/ -Hsrc/OSPSuite.FuncParserNative/ -DCMAKE_BUILD_TYPE=Release -DCMAKE_OSX_ARCHITECTURES=x86_64
```

Comment:
I see, separate lines per arch. Do you know this is required? Cmake should know what platform to build for.